React-Calculator/src/components/WolframCalc.jsx

```diff
@@ -28,81 +28,92 @@ const getApiData = async (number1, operator, accumulator) => {
     const [number1, setNumber1] = useState(null);
     const [operator, setOperator] = useState(null);
     const [accumulator, setAccumulator] = useState('');
 
     async function addData (input) {
-        if (!isOff) {
-            if (operators.includes(input)) {
-                if (!number1) {
-                    setNumber1(accumulator);
-                    setAccumulator('');
-                    setOperator(input);
-                }
-            } else {
-                switch (input) {
-                    case '=':
-                        const apiData = await getApiData(number1, operator, accumulator);
-                        setAccumulator(apiData);
-                        setLogList((previousLogs) => {
-                            const newLogs = {
-                                calculo: `${number1} ${operator} ${accumulator}`,
-                                date: new Date(),
-                            }
-                            return [...previousLogs, newLogs]
-                        }
-                        )
-                        break;
-                    case 'C':
-                        setAccumulator('');
-                        setNumber1(null);
-                        setOperator(null);
-                        break;
-                    case 'del':
-                        if ((number1 && operator && accumulator) || accumulator) {
-                            setAccumulator(accumulator.substring(0, accumulator.length - 1));
-                        } else if (operator) {
-                            setOperator(operator.substring(0, operator.length - 1));
-                        } else if (number1) {
-                            setNumber1(number1.substring(0, number1.length - 1));
-                        }
-                        break;
-                    case 'Off':
-                        setIsOff(true);
-                        break;
-                    case '.':
-                        if (accumulator.includes('.')) {
-                            setAccumulator(accumulator);
-                        } else {
-                            setAccumulator(accumulator + input);
-                        }
-                        break;
-                    default:
-                        if(accumulator.length < 28) setAccumulator(accumulator + input);
-                }
-            }
-        } else if (input === 'Off') {
-            setAccumulator('');
-            setNumber1(null);
-            setOperator(null);
-            setIsOff(false);
+        if(isOff && input !== 'Off'){
+            return;
         }
+
+        if (operators.includes(input) && !number1) {
+                setNumber1(accumulator);
+                setAccumulator('');
+                setOperator(input);
+                return;
+        }
+
+        refactoredSwitch({ input, setAccumulator, accumulator, setNumber1, number1, setOperator, operator, setIsOff, isOff ,setLogList })
```

Comment:
not the prettiest, most descriptive name (in terms of what it does, at least) for a function `refactoredSwitch`, I know you agree with me because we discussed it, let's fix it for the next one